internal/manager/telemetry/manager_test.go

```diff
@@ -27,6 +27,12 @@ import (
 	gatewayv1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
 )
 
+type mockGatewaysCounter struct{}
```

Comment:
Nit: might as well make it 

```suggestion
type mockGatewaysCounter int
```

and report the underlying `int` in `GatewayClientsCount()`. Will make less of a surprise for the caller when using it as: `mockGatewaysCounter(5)` instead of `mockGatewaysCounter{}`

Reply:
👍 [88dad8f](https://github.com/Kong/kubernetes-ingress-controller/pull/3783/commits/88dad8f6ae05b599a80b2f0eca0f5fd893c2d226)